### PR TITLE
Android 12 Updates v3

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

- Upgrade `targetSdkVersion` and `compileSdkVersion` to API 31
- Bump `browser-switch` version to `1.X.X` [TO DO - pending browser switch release]

**Note**: Manually tested all flows except visa checkout does not seem to be working in the demo app before or after these changes.

 ### Checklist

 - [x] Added a changelog entry

- @sarahkoop

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Braintree Android SDK Release Notes
 
+## unreleased
+
+* Upgrade `targetSdkVersion` and `compileSdkVersion` to API 31
+* Bump `browser-switch` version to `2.X.X`
```
Comment (scannillo) [inline CHANGELOG.md:6]: Looks like it is still `api 'com.braintreepayments:browser-switch:1.1.3'` in this PR?

Comment (scannillo) [reply CHANGELOG.md:6]: Should we hold on this CHANGELOG entry for now?

**Review by scannillo — APPROVED** [review]
Looks good! One small comment about CHANGELOG

File: CHANGELOG.md (hunk 1 above)
Comment (sarahkoop) [reply CHANGELOG.md:6]: Yes this is pending the browser switch release with Android 12 support - will update the dependency and CHANGELOG after that release

**Pull request merged**